### PR TITLE
Add rule to check for manual date formatting

### PR DESCRIPTION
**Pull request by ehoogerbeets**
- engineer should use a date formatting library instead because dates and times are locale-dependent, not just language-dependent.
- Also, translators often do not know what to do with formats like that, so they just leave them alone. They are linguists, not programmers. Using a library avoids that.

### CODE REVIEW

File: docs/source-no-manual-currency-formatting.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# source-no-manual-currency-formatting
```
Comment (wadimw) [inline docs/source-no-manual-currency-formatting.md:1]: merge base is incorrect

File: src/plugins/BuiltinPlugin.js [hunk 2]
```diff
@@ -164,6 +164,66 @@ export const regexRules = [
...
+        link: "https://github.com/ilib-js/i18nlint/blob/main/docs/source-no-manual-currency-formatting.md",
+        severity: "error"
+    },
+    {
+        type: "resource-source",
+        name: "source-no-manual-date-formatting",
+        description: "Ensure that source strings do not contain manually formatted dates or times. Dates and times should be formatted using a locale-sensitive date formatter instead.",
+        note: "Do not format dates or times in English strings. Use a locale-sensitive date formatter and substitute the result of that into this string.",
+        regexps: [
+            "\\{years?\\}[/\\-\\. ]\\{months?\\}[/\\-\\. ]\\{days?\\}",
```
Comment (wadimw) [inline src/plugins/BuiltinPlugin.js:193]: To increase readability (mostly get rid of all those extra `\`), we could do it like
```js
/\{years?\}[/\-\. ]\{months?\}[/\-\. ]\{days?\}/.source
```

Comment (wadimw) [reply src/plugins/BuiltinPlugin.js:193]: Or update declarative rule loading to accept `string | RegExp`

File: src/rules/DeclarativeResourceRule.js [hunk 3]
```diff
@@ -139,9 +139,11 @@ class DeclarativeResourceRule extends ResourceRule {
             }
         }
         let results = [];
-        this.re.forEach(re => {
+        // only need 1 regexp to match in order to trigger this rule
+        for (const re of this.re) {
             results = results.concat(this.checkString({re, source, target, file, resource}));
-        });
+            if (results.length > 0) break;
```
[the remark this answers is not in the document]
Comment (wadimw) [reply src/rules/DeclarativeResourceRule.js:145]: Maybe we could add a property for declarative rules where we would toggle between "any" and "all" (i.e. `regexps.some()` and `regexps.map()`)

File: src/plugins/BuiltinPlugin.js (hunk 2 above)
Comment (ehoogerbeets) [reply src/plugins/BuiltinPlugin.js:193]: We should do that in a separate PR. This PR is about the date formatting.

File: src/rules/DeclarativeResourceRule.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ehoogerbeets) [reply src/rules/DeclarativeResourceRule.js:145]: Same

[the remark this answers is not in the document]
Comment (wadimw) [reply src/rules/DeclarativeResourceRule.js:145]: Needs docs update

**Review by wadimw — APPROVED** [review]

**Pull request merged**